int128/groovy-ssh
# Fix adding host key for gateway access

This fixes the bug of #198.

Bug1: The knownHosts file is modified even if `knownHosts = file(...)` when using the gateway access.

Bug2: A tunnel host and port is added to the knownHosts file if `knownHosts = addHostKey(file(...))` when using the gateway access. It should be the remote host and port.

### Steps to verify the fix
1. Connect to a remote host via a gateway host with `knownHosts = file(...)`.
2. Verify that the knownHosts file is not modified.

1. Connect to a remote host via a gateway host with `knownHosts = addHostKey(file(...))`.
2. Verify that the knownHosts file contains the remote host and port, not the tunnel host and port.


### Backward compatibility
No breaking changes.
